src/app/application-management/application-detail/application-detail-resolver.service.ts

```diff
@@ -19,31 +19,57 @@ import { Csar } from '../../core/model/csar.model';
 import { forkJoin, Observable, of } from 'rxjs';
 import { catchError, mergeMap } from 'rxjs/operators';
 import { Plan } from '../../core/model/plan.model';
+import { Interface } from '../../core/model/interface.model';
+import { PlanTypes } from '../../core/model/plan-types.model';
+import { select } from '@angular-redux/store';
 
 @Injectable()
-export class ApplicationDetailResolverService implements Resolve<{ csar: Csar, buildPlan: Plan, terminationPlan: Plan }> {
+export class ApplicationDetailResolverService implements Resolve<{ csar: Csar, buildPlan: Plan, terminationPlan: Plan, interfaces: Interface[] }> {
```

Comment:
Can't we get rid of buildPlan and terminationPlan?
I suggest that we fetch all plans, i.e. build and termination plans, via boundary definitions into the interfaces array. Of course, this has to be adapted in application-detail.component accordingly.

Reply:
We could get rid of these here. However, if there is no build plan available, we have to notify the user somehow. We have to decide where we want to resolve the incoming data. In the Resolver or the Component.